### PR TITLE
helanal cleanup

### PR DESCRIPTION
**Pull request by orbeckst**
- fixed broken helanal_trajectory() and helanal_main()
- slightly better tests
- minor code cleanups

### CODE REVIEW

**Comment by orbeckst** [comment]
This is done (I am not addressing the QC issues because I actually do not want to add doc strings to `test_something()` tests as the doc string would replace the rather useful default output of `nosetests -v `).

**Comment by orbeckst** [comment]
Pending favorable reviews I'd like this to go into 0.13.0 (#600).

File: package/MDAnalysis/analysis/helanal.py [hunk 1]
```diff
@@ -129,41 +129,27 @@ def center(coordinates):
...
-vecscaler = np.dot
-
+veclength = mdamath.norm
 
 def vecnorm(a):
     """Return a/|a|"""
     return a / veclength(a)
 
-
 def vecangle(a, b):
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/helanal.py]: mdamath.angle already covers nan issues, so this function can disappear (find and replace vecangle -> mdamath.angle)

File: package/MDAnalysis/analysis/helanal.py [hunk 2]
```diff
@@ -129,41 +129,27 @@ def center(coordinates):
...
-
-def veclength(v):
-    """Length of vector *v*."""
-    # note: this is 3 times faster than np.linalg.norm
-    return np.sqrt(np.dot(v, v))
-
-
-vecscaler = np.dot
-
+veclength = mdamath.norm
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/helanal.py]: Find and replace veclength -> mdamath.norm?

File: package/MDAnalysis/analysis/helanal.py [hunk 3]
```diff
@@ -129,41 +129,27 @@ def center(coordinates):
...
     (instead of `nan`).
     """
-    angle = np.arccos(np.dot(a, b) / (veclength(a) * veclength(b)))
+    angle = mdamath.angle(a, b)
     if np.isnan(angle):
         return 0.0
     return angle
 
-
 def vecdist(a, b):
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/helanal.py]: This function is never actually used?

**Comment by richardjgowers** [comment]
I think I'm just being pedantic, if you want to merge just go for it

**Comment by orbeckst** [comment]
Pedantic is good. Will fix. 

--
Oliver Beckstein
email: orbeckst@gmail.com


Am Jan 7, 2016 um 4:17 schrieb Richard Gowers <notifications@github.com>:

**Comment by orbeckst** [comment]
Addressed all your comments and rebased against develop. Good to go (assuming travis is happy).

**Comment by orbeckst** [comment]
@richardjgowers — all good.

**Pull request merged**